pkg/virt-launcher/virtwrap/network/cloud-init-network_test.go

```diff
@@ -0,0 +1,225 @@
+// Inspired by pkg/virt-launcher/virtwrap/network/podinterface_test.go
+
+package network
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net"
+	"os"
+	"strconv"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/vishvananda/netlink"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"gopkg.in/yaml.v2"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/log"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+var _ = Describe("cloud-init network", func() {
+	var mockNetwork *MockNetworkHandler
+	var ctrl *gomock.Controller
+
+	var tmpDir string
+
+	log.Log.SetIOWriter(GinkgoWriter)
+
+	BeforeEach(func() {
+		tmpDir, _ := ioutil.TempDir("", "cloudinittest")
+		setInterfaceCacheFile(tmpDir + "/cache-%s.json")
+
+		ctrl = gomock.NewController(GinkgoT())
+		mockNetwork = NewMockNetworkHandler(ctrl)
+		Handler = mockNetwork
+	})
+
+	AfterEach(func() {
+		os.RemoveAll(tmpDir)
+	})
+
+	Context("on successful GenNetworkFile()", func() {
+		It("should create valid network-config contents", func() {
+			count := 3
+			domain := newSriovDomainWithInterface()
```

Comment:
At one point, I was told by other reviewers that I should use ```SRIOV``` (all caps) in function names. I don't personally insist since I find this convention a bit confusing, but just a heads up for you that others may feel differently.

Reply:
I'm open to any renames, I bounced back and forth for readability. I'm not even sure i like `GenNetworkFile`.